src/canonmn_int.cpp

```diff
@@ -2851,6 +2850,33 @@ std::ostream& printCsLensFFFF(std::ostream& os, const Value& value, const ExifDa
   return EXV_PRINT_TAG(canonCsLensType)(os, value, metadata);
 }
 
+/**
+ * @brief convert string to float w/o considering locale
+ *
+ * Using std:stof to convert strings to float takes into account the locale
+ * and thus leads to wrong results when converting e.g. "5.6" with a DE locale
+ * which expects "," as decimal instead of ".". See GitHub issue #2746
+ *
+ * Use std::from_chars once that's properly supported by compilers.
+ *
+ * @param str string to convert
+ * @return float value of string
+ */
+float string_to_float(std::string const& str) {
+  float val{};
+  std::stringstream ss;
+  std::locale c_locale("C");
+  ss.imbue(c_locale);
+  ss << str;
+  ss >> val;
+
+  if (ss.fail()) {
+    throw Error(ErrorCode::kerErrorMessage, std::string("canonmn_int.cpp:string_to_float failed for: ") + str);
+  }
```

Comment:
```suggestion
  float val{};
#ifdef __cpp_lib_to_chars
  auto [ptr, ec] = std::from_chars(str.data(), str.data() + str.size(), val);
#else
  std::stringstream ss;
  std::locale c_locale("C");
  ss.imbue(c_locale);
  ss << str;
  ss >> val;
#endif

#ifdef __cpp_lib_to_chars
  if (ec != std::errc()) {
#else
  if (ss.fail()) {
#endif
    throw Error(ErrorCode::kerErrorMessage, std::string("canonmn_int.cpp:string_to_float failed for: ") + str);
  }
```

Reply:
Will need this snippet at the top of the file as well:
```
#if __has_include(<charconv>)
#  include <charconv>
#endif
```

Reply:
I'd rather leave the code as is until we can fully move over. This doubles the amount of code, increases cognitive load, and we would need to ensure that we actually test this with new enough compilers in CI. And there isn't any real benefit since this isn't performance critical.